spanner/internal/testutil/inmem_spanner_server.go

```diff
@@ -681,48 +681,57 @@ func (s *inMemSpannerServer) getStatementResult(sql string) (*StatementResult, e
 
 func (s *inMemSpannerServer) simulateExecutionTime(method string, req interface{}) (interface{}, error) {
 	s.mu.Lock()
-	defer s.mu.Unlock()
```

Comment:
Keeping this lock here prevented the server from being shut down if there was a request still waiting for an execution time to finish (e.g. if an execution time of 1 minute had been set for an RPC, then keeping this lock would prevent the tests from stopping the server for 1 minute, which again would mean that the test would take 1 minute).